### PR TITLE
Alerting: Support provisioning of time intervals

### LINKED ISSUE #83168: Alerting: Grafana Managed Alerts should use time_intervals instead of mute_time_intervals
Grafana Managed Alerts should use the new `time_intervals` field instead of `mute_time_intervals`, which is deprecated.

### Backend

In the backend, this requires changes to `grafana/alerting` and `grafana/grafana`. The `Configuration` interface in `grafana/alerting` will need `TimeIntervals()`:

```go
type Configuration interface {
	DispatcherLimits() DispatcherLimits
	InhibitRules() []InhibitRule
	TimeIntervals() []TimeInterval
	MuteTimeIntervals() []MuteTimeInterval
	...
}
```

and the initialization of mute timings will need to use both `TimeIntervals()` and `MuteTimeIntervals()` method:

```go
am.timeIntervals = am.buildTimeIntervals(cfg.TimeIntervals(), cfg.MuteTimeIntervals())
```

In `grafana/grafana`, the Grafana API will need to be updated for both the UI and Provisioning to use `time_intervals` instead. We should also write a migration to move `mute_time_intervals` to `time_intervals`.

### Frontend

The frontend will need to be updated to show `time_intervals` in the UI, and use them whenever creating or editing an existing time interval. Mute timings should also be renamed to Time intervals.

### PR DESCRIPTION
**Pull request by grobinson-grafana**
**What is this feature?**

This PR adds support for provisioning time intervals using API and File provisioning. It does not add support for the export endpoints. These will be added in a subsequent PR. We added support for time intervals to the Grafana API in https://github.com/grafana/grafana/pull/83216.

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #83168

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [x] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/services/ngalert/provisioning/time_intervals_test.go [hunk 1]
```diff
@@ -0,0 +1,661 @@
+package provisioning
```
Comment (grobinson-grafana) [inline pkg/services/ngalert/provisioning/time_intervals_test.go:1]: This file was copied from `mute_timings_test.go`.

**Comment by yuri-tceretian** [comment]
I do not think we need one more service for time-intervals: 1. it causes confusion which one is supposed to be used. 2. It creates one more API and therefore requires one more resource.  I think we just need to update the existing service to handle. The provsioning API already abstracts Alertmanager configuration and therefore, we do not have to keep both APIs for migration. It can be done transparently.

File: pkg/services/ngalert/api/tooling/definitions/provisioning_time_intervals.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+// swagger:parameters RoutePostTimeInterval RoutePutTimeInterval
+type TimeIntervalHeaders struct {
+	// in:header
+	XDisableProvenance string `json:"X-Disable-Provenance"`
+}
+
+// swagger:model
+type TimeInterval struct {
+	config.TimeInterval `json:",inline" yaml:",inline"`
```
Comment (yuri-tceretian) [inline pkg/services/ngalert/api/tooling/definitions/provisioning_time_intervals.go:76]: I think we should not expose the underlying models going forward.  There are already API models for TimeInterval.
https://github.com/grafana/grafana/blob/2ab7d3c725969927434cb7af71c6cdba9fc21df7/pkg/services/ngalert/api/tooling/definitions/time_intervals.go#L40-L64

**Comment by yuri-tceretian** [comment]
Recently, we have introduced a new APIs https://github.com/grafana/grafana/pull/81672 called time-intervals, which will replace the provisioning API in the future and naturally remove the provisioning one. The new API will be used everywhere: in UI, in Terraform and Go client. 
Currently, it uses the mute-timing service as the backend, and I think we just need to update the backend service to write to a new destination.

**Comment by grobinson-grafana** [comment]
Discussed offline, we will do something else.

**Pull request closed without merging**